### DISCUSSION TITLE
There seems to be a problem with using one large quickstart repo

### DISCUSSION THREAD

**Discussion by starksm64** [quarkusio/quarkus#44859] [Design Discussions]
Today I tried running a quick start in two cloud environments:
1. Heroku. Going through the https://quarkus.io/guides/deploying-to-heroku the initial push to heroku (after fixing the steps as outlined in https://github.com/quarkusio/quarkus/issues/44857) fails because the entire https://github.com/quarkusio/quarkus-quickstarts.git repo is built rather than just the getting-started submodule. The resulting content exceeds the allowed slug size.
2. OpenShift DevSpaces. I selected the 'Quarkus REST API' sample, and it launches a bit slowly into this view:
![devspace_launch](https://github.com/user-attachments/assets/3dee0532-6f8d-44f0-bfd1-f9d183d1eb55)

I have to know that the REST Api sample corresponds to the rest-client-quickstarts folder, and then when I navigate to that folder and select the ExtensionsResourceTest and select run, the devspace env is again building the entire quarkus-quickstarts. It is taking several minutes so far, and I'm doubtful the build will succeed at this point.

We are wasting a ton of resources and time by pulling in the entire quarkus-quickstarts repo as the source development repo for these environments. What I had to do for heroku was to copy out the getting-started submodule directory and reinitialize a new git repo and push that. We need to be doing the same for these devspaces samples.

**Comment by cescoffier** [comment]
That repo was not designed to be used a one big thing. 
Can't you set the "context path" in these cloud environment to only select the directory you need?

**Comment by starksm64** [comment]
This is how the current guides and whatever the devspaces button is doing
are using the quickstart repo. I'm just pointing out that if cloud
environments that integrate with a github repo expect that it represents
the application, it is not going to work well with the quickstarts repo,
and guides should be specifying how to build a single sample application.

**Reply by cescoffier** [reply]
Every guide indicates that you must navigate to a specific directory to run the sample application (hopefully, if not, that's a bug).

BTW, I don't see the devspace button. Where do you see it?

**Reply by starksm64** [reply]
The https://quarkus.io/guides/deploying-to-heroku definitely does not, and the real problem there is that it is the cloud env responsible for doing the build, and it is just treating the repo as the application source that needs to be built.

When you create a workspace, this screen is shows and has a "Quarkus REST API" sample button:
![CreateWorkspace](https://github.com/user-attachments/assets/76bc950b-711e-4f62-94d9-540692bc399b)

Aside, why is there a "Java Lombok" sample as a top item?

**Reply by cescoffier** [reply]
Thanks for the report on the Heroku guide. I will log an issue to fix it.

I was not aware of the Dev Spaces "template" (they are not managed by us). We can definitely improve (and no, I've no idea what Java Lombok is, and as I really dislike Lonbok, I'm a bit concerned). @maxandersen, are you aware of these templates?

**Comment by starksm64** [comment]
There are several issues with the current heroku guide which I have
captured in this issue:
https://github.com/quarkusio/quarkus/issues/44857

I'll create a PR for that
